tests/transactionScheduler_events_test.cdc

```diff
@@ -0,0 +1,261 @@
+    Test.assert(pendingExecutionEventsBeforeTime.length == 0, message: "PendingExecution before time")
+
+    // move time forward to trigger execution eligibility
+    // Have to subtract to handle the automatic timestamp drift
+    // so that the medium transaction that got scheduled doesn't get marked as pendingExecution
+    Test.moveTime(by: Fix64(futureDelta - 6.0))
+    while getTimestamp() < timeInFuture {
+        Test.moveTime(by: Fix64(1.0))
+    }
+
+    // Simulate FVM process - should process since timestamp is in the past
+    processTransactions()
+
+    // Check for PendingExecution event after processing
+    // Should have one high
+
+    let pendingExecutionEventsAfterTime = Test.eventsOfType(Type<FlowTransactionScheduler.PendingExecution>())
+    Test.assertEqual(1, pendingExecutionEventsAfterTime.length)
+    
+    var i = 0
+    var firstEvent: Bool = false
+    for event in pendingExecutionEventsAfterTime {
+        let pendingExecutionEvent = event as! FlowTransactionScheduler.PendingExecution
+        Test.assert(
+            pendingExecutionEvent.id != UInt64(2),
+            message: "ID 2 Should not have been marked as pendingExecution"
+        )
+
+        // verify that the transactions got marked as executed
+        var status = getStatus(id: pendingExecutionEvent.id)
+        Test.assertEqual(statusExecuted, status!)
+
+        // Simulate FVM execute - should execute the transaction
+        if pendingExecutionEvent.id == transactionToFail {
+            // ID 2 should fail, so need to verify that
+            executeScheduledTransaction(id: pendingExecutionEvent.id, testName: "Test Transaction Execution: First High Scheduled", failWithErr: "Transaction \(transactionToFail) failed")
+        } else {
+            executeScheduledTransaction(id: pendingExecutionEvent.id, testName: "Test Transaction Execution: First High Scheduled", failWithErr: nil)
+        
+            // Verify that the first event is the high priority transaction
+            if !firstEvent {
+                let executedEvents = Test.eventsOfType(Type<FlowTransactionScheduler.Executed>())
+                Test.assert(executedEvents.length == 1, message: "Should only have one Executed event")
+                let executedEvent = executedEvents[0] as! FlowTransactionScheduler.Executed
+                Test.assertEqual(pendingExecutionEvent.id, executedEvent.id)
+                Test.assertEqual(pendingExecutionEvent.priority, executedEvent.priority)
+                Test.assertEqual(pendingExecutionEvent.executionEffort, executedEvent.executionEffort)
+                Test.assertEqual(pendingExecutionEvent.transactionHandlerOwner, executedEvent.transactionHandlerOwner)
+                Test.assertEqual(pendingExecutionEvent.transactionHandlerTypeIdentifier, executedEvent.transactionHandlerTypeIdentifier!)
+                firstEvent = true
+            }
+        }
+
+        i = i + 1
```

Comment:
why do we need this? I think it would be wise to check i == 1 (I believe)
Epsecially it would be very good to check the ID of transaction to be executed and match it to the ID from the above schedule transaction, because this is critical, if these IDs for some reason get mixed it will cause critical failure.